<?xml version="1.0"?>

<launch>
    <arg name="world_name" default="$(find unitree_move_base)/worlds/hospital.world"/>
    <arg name="map_file" default="hospital"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <!-- <arg name="world_name" value="$(arg world_name)"/> -->
        <arg name="world_name" value="$(find aws_robomaker_hospital_world)/worlds/hospital.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="false"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
        <arg name="verbose" default="true"/>
    </include>

    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(find scout_bringup)/maps/$(arg map_file).yaml" output="screen"/>

    <!--scout 1-->
    <include file="$(find scout_bringup)/launch/scout_mini_bringup.launch">
        <arg name="ns" value="robot_1"/>
        <arg name="x" value="2.0"/>
        <arg name="y" value="10.0"/>
        <arg name="z" value="0.5"/>
        <arg name="yaw" value="0.0"/>
    </include>

    <include file="$(find scout_bringup)/launch/scout_mini_bringup.launch">
        <arg name="ns" value="robot_2"/>
        <arg name="x" value="2.0"/>
        <arg name="y" value="12.0"/>
        <arg name="z" value="0.5"/>
        <arg name="yaw" value="0.0"/>
    </include>

    <include file="$(find scout_bringup)/launch/scout_mini_bringup.launch">
        <arg name="ns" value="robot_3"/>
        <arg name="x" value="4.0"/>
        <arg name="y" value="10.0"/>
        <arg name="z" value="0.5"/>
        <arg name="yaw" value="0.0"/>
    </include>

    <include file="$(find scout_bringup)/launch/scout_mini_bringup.launch">
        <arg name="ns" value="robot_4"/>
        <arg name="x" value="4.0"/>
        <arg name="y" value="12.0"/>
        <arg name="z" value="0.5"/>
        <arg name="yaw" value="0.0"/>
    </include>

    <!--go2 1-->
    <include file="$(find unitree_guide)/launch/go2_bringup.launch">
        <arg name="ns" value="robot_5"/>
        <arg name="initial_x" value="-2.0"/>
        <arg name="initial_y" value="10.0"/>
        <arg name="initial_z" value="0.5"/>
    </include>

    <include file="$(find unitree_guide)/launch/go2_bringup.launch">
        <arg name="ns" value="robot_6"/>
        <arg name="initial_x" value="-2.0"/>
        <arg name="initial_y" value="12.0"/>
        <arg name="initial_z" value="0.5"/>
    </include>

    <node pkg="rviz" type="rviz" name="rviz_move_base" args="-d $(find scout_bringup)/rviz/multi_mini.rviz"/>
    
</launch>
